### PR TITLE
feat: improve Appium iOS tests speed

### PR DESCRIPTION
**Pull request by igorgiumellizup**
### Related Issues

#1500 

### Description and Example

- Improved test scenarios (ActionNotRegistered, Alert, AddChildren, Button, Conditional and Touchable). More tests will be improved on future PRs

- Improved element locator strategy for iOS. 

iOS-comparisson
android-comparisson

### Checklist

Please, check if these important points are met using `[x]`:

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.
- [x] I have made the documentation changes or I created an issue explaining how to document this change on [Docs issues](https://github.com/ZupIT/beagle-docs/issues). Added a notice at the XPath section, commenting about native locators on iOS: https://github.com/ZupIT/beagle/wiki/Mobile-tests-with-Appium#xpath-element-attributes

[PR Guide]: https://github.com/ZupIT/beagle/blob/master/doc/contributing/pull_requests.md

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt [hunk 1]
```diff
@@ -72,52 +72,56 @@ class TextInputScreenSteps : AbstractStep() {
...
+    fun checkKeyboardFocus(elementText: String) {
+        val mobileElement = waitForElementWithValueToBeClickable(elementText,
+            likeSearch = false,
+            ignoreCase = false,
+            nativeLocator = false
+        )
+        Assert.assertTrue(elementText.equals(mobileElement.text))
         mobileElement.sendKeys("a")
-        Assert.assertFalse(string.equals(mobileElement.text))
+        Assert.assertFalse(elementText.equals(mobileElement.text))
```
Comment (carlossteinzup) [inline tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt:83]: Será que aqui ficaria mais adequado se fosse "textElement"? na linha 83?

File: tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt [hunk 2]
```diff
@@ -72,52 +72,56 @@ class TextInputScreenSteps : AbstractStep() {
...
         mobileElement.clear()
-        Assert.assertTrue(string.equals(mobileElement.text))
+        Assert.assertTrue(elementText.equals(mobileElement.text))
```
Comment (carlossteinzup) [inline tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt:85]: Será que aqui ficaria mais adequado se fosse "textElement"? na linha 85?

**Review by carlossteinzup — APPROVED** [review]

File: tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt (hunk 1 above)
Comment (igorgiumellizup) [reply tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt:83]: Opa, elementText = texto do elemento, textElement = elemento do tipo texto
Ali é pra ser só o texto do elemento

File: tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt (hunk 2 above)
Comment (igorgiumellizup) [reply tests/appium/project/src/test/kotlin/br/com/zup/beagle/cucumber/steps/TextInputScreenSteps.kt:85]: Opa, elementText = texto do elemento, textElement = elemento do tipo texto
Ali é pra ser só o texto do elemento

**Review by carlossteinzup — APPROVED** [review]

**Review by dantes-git — APPROVED** [review]

**Pull request merged**